test/sharness/t0182-circuit-relay.sh

```diff
@@ -50,7 +56,14 @@ test_expect_success 'connect B <-> Relay' '
 '
 
 test_expect_success 'wait until relay is ready to do work' '
-  sleep 1
+  while ! ipfsi 2 swarm connect /p2p/$PEERID_1/p2p-circuit/p2p/$PEERID_0; do
+    iptb stop &&
+    iptb_wait_stop &&
+    iptb start -wait -- --routing=none &&
+    iptb connect 0 1 &&
+    iptb connect 2 1 &&
+    sleep 5
+  done
```

Comment:
Tests in  `go-ipfs/test/sharness/t0182-circuit-relay.sh`   are flaky.

`ipfsi 2 swarm connect /p2p/$PEERID_1/p2p-circuit/p2p/$PEERID_0`   sometimes fails with:
```
NO_RESERVATION (204)
Error: connect 12D3KooWLuY1Q13o6Q91NoJn4Qv7RSq5rYS6uC4YCzKCJ3S8GnwQ failure: failed to dial 12D3KooWLuY1Q13o6Q91NoJn4Qv7RSq5rYS6uC4YCzKCJ3S8GnwQ:
  * [/p2p/12D3KooWNfs8uFpQf6NnseZKLyS3c8EFppGtUGejHpViBeg8f7Vt/p2p-circuit] error opening relay circuit: NO_RESERVATION (204)`
```

- it happens randomly, bumping it to 10 seconds does not help much, it still fails sometimes.
- shutting down nodes and starting them again fixes the problem
  -  it feels like covering up some underlying racy bug:   once `NO_RESERVATION` error is returned, the relay will never work (until reboot of entire testbed).